File: src/Employee.py
import random


class Employee:
    name = ""
    lastname = ""
    birthday = ""
    date = ""
    rfc = ""
    curb = ""
    schedule = ""
    rol = ''
    salary = 1
    id = 1

    def __init__(self, name, lastname, birthday, date, rfc, curb, rol, salary, schedule):
        self.name = name.upper()
        self.lastname = lastname.upper()
        self.birthday = birthday
        self.date = date
        self.rfc = rfc.upper()
        self.curb = curb.upper()
        self.salary = salary
        self.rol = rol
        self.id = random.randint(0, 10000)
        self.schedule = schedule

    def showData(self):
        if (self.rol == 'v'):
            self.job="Veterinario"
        elif(self.rol=='g'):
            self.job="Guía"
        elif(self.rol=='a'):
            self.job="Administración"
        elif(self.rol=='m'):
            self.job="Mantenimiento"
        print(
              f"El id del empleado es   {self.id}  y  es {self.name}  {self.lastname} , su labor es {self.job} , su RFC es  {self.rfc} , su curp es  {self.curb} , su salario es  {self.salary} , su fecha de entrada es {self.date} , su fecha de nacimiento es  {self.birthday}")

    # setters y getters
    def setName(self, name):
        self.name = name.upper()

    def getName(self):
        return self.name

    def setLastname(self, lastname):
        self.lastname = lastname.upper()

    def getLastname(self):
        return self.lastname

    def setBirthday(self, birthday):
        self.birthday = birthday

    def getBirthday(self):
        return self.birthday

    def setCurb(self, curb):
        self.curb = curb.upper()

    def getCurb(self):
        return self.curb

    def setDate(self, date):
        self.date = date

    def getDate(self):
        return self.date

    def setRfc(self, rfc):
        self.rfc = rfc.upper()

    def getRfc(self):
        return self.rfc

    def setSalary(self, salary):
        self.salary = salary

    def getSalary(self):
        return self.salary

    def setRol(self, rol):
        self.rol = rol

    def getRol(self):
        return self.rol

    def getSchedule(self):
        return self.schedule

    def setSchedule(self, schedule):
        self.schedule = schedule

    def setId(self, id):
        self.id = id

    def getId(self):
        return self.id

    def changeId(self):
        self.id = random.randint(0, 10000)

File: src/Employee.java
import java.util.Random;

public class Employee {
    Random ran = new Random();
    private String name, lastname, birthday, date, rfc, curb, schedule;
    private char rol;
    private double salary;
    private int id;

    public Employee(String name, String lastname, String birthday, String date, String rfc, String curb, char rol, double salary, String schedule) {
        this.name = name.toUpperCase();
        this.lastname = lastname.toUpperCase();
        this.birthday = birthday;
        this.date = date;
        this.rfc = rfc.toUpperCase();
        this.curb = curb.toUpperCase();
        this.salary = salary;
        this.rol = rol;
        this.id = ran.nextInt(100000);
        this.schedule = schedule;
    }

    public void showData() {
        String job = "";
        switch (this.rol){
            case 'v': job = "Veterinario";
                break;
            case 'g': job = "Guía";
                break;
            case 'm': job = "Mantenimiento";
                break;
            case 'a': job = "Administración";
                break;
        }
        System.out.println("El id del empleado es " + this.id + ", y  es " + this.name + " " + this.lastname + ", su labor es " + job + ", su RFC es " + this.rfc + ", su curp es " + this.curb + ", su salario es " + this.salary + ", su fecha de entrada es " + this.date + ", su fecha de nacimiento es " + this.birthday);

    }

    //setters y getters
    public void setName(String name) {
        this.name = name.toUpperCase();
    }

    public String getName() {
        return name;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname.toUpperCase();
    }

    public String getLastname() {
        return lastname;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setCurb(String curb) {
        this.curb = curb.toUpperCase();
    }

    public String getCurb() {
        return curb;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getDate() {
        return date;
    }

    public void setRfc(String rfc) {
        this.rfc = rfc.toUpperCase();
    }

    public String getRfc() {
        return rfc;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    public double getSalary() {
        return salary;
    }

    public void setRol(char rol) {
        this.rol = rol;
    }

    public char getRol() {
        return rol;
    }

    public String getSchedule() {
        return schedule;
    }

    public void setSchedule(String schedule) {
        this.schedule = schedule;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public void changeId() {
        this.id = ran.nextInt(100000);
    }
}